### PR TITLE
HDDS-8237. [Snapshot] loadDb() used by SstFiltering service creates extraneous directories.

### PR DESCRIPTION
**Pull request by GeorgeJahad**
## What changes were proposed in this pull request?

Fix bug that creates extraneous snapshot directories
```
ls -ld /data/metadata/db.snapshots/db*
drwxr-xr-x 2 hadoop hadoop 4096 Mar 22 03:27 /data/metadata/db.snapshots/db.checkpoints
drwxr-xr-x 2 hadoop hadoop 4096 Mar 22 03:27 /data/metadata/db.snapshots/db.snapshots
```
## What is the link to the Apache JIRA
https://issues.apache.org/jira/browse/HDDS-8237

## How was this patch tested?

Ran 'ls' again as above.

### CODE REVIEW

**Review by hemantk-12 — APPROVED** [review]
Thanks for the patch @GeorgeJahad.

+1.

Left couple of minor comments.

**Review by hemantk-12 — APPROVED** [review]

File: hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/OmMetadataManagerImpl.java [hunk 1]
```diff
@@ -484,14 +484,23 @@ public void start(OzoneConfiguration configuration) throws IOException {
...
   }
 
   public static DBStore loadDB(OzoneConfiguration configuration, File metaDir,
                                String dbName, boolean readOnly,
                                java.util.Optional<Boolean>
                                        disableAutoCompaction)
           throws IOException {
+    return loadDB(configuration, metaDir, dbName, readOnly,
+        disableAutoCompaction, true);
+  }
```
Comment (prashantpogde) [inline hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/OmMetadataManagerImpl.java:497]: nit : an empty line between functions.

**Review by prashantpogde — APPROVED** [review]
LGTM with a minor nit.

File: hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/OmMetadataManagerImpl.java (hunk 1 above)
Comment (GeorgeJahad) [reply hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/OmMetadataManagerImpl.java:497]: fixed

**Comment by prashantpogde** [comment]
@GeorgeJahad Can you please resolve the conflict ?

**Review by smengcl — APPROVED** [review]
Thanks @GeorgeJahad for taking care of this.

**Comment by smengcl** [comment]
CI passed on @GeorgeJahad 's fork's branch: https://github.com/GeorgeJahad/ozone/actions/runs/4613253792

But there are some (seemingly irrelevant) flakiness in PR CI:

https://github.com/apache/ozone/actions/runs/4613253988/jobs/8155048388
```
Error:  Failures: 
Error:    TestContainerStateCounts.testDeletedContainerCount:383 expected: <5> but was: <2>
```

https://github.com/apache/ozone/actions/runs/4613253988/jobs/8155259011
```
Error:  Failed to execute goal org.apache.hadoop:hadoop-maven-plugins:3.3.5:version-info (version-info) on project ozone-common: java.util.ConcurrentModificationException -> [Help 1]
```

Retriggered PR CI.

**Comment by smengcl** [comment]
Thanks @hemantk-12 and @prashantpogde for reviewing this.

**Pull request merged**